gradio/components.py

```diff
@@ -2758,8 +2766,11 @@ def postprocess(self, y: str | pd.DataFrame) -> Dict:
         if y is None:
             return None
         if isinstance(y, str):
-            y = pd.read_csv(y)
-            return {"headers": y.columns.values.tolist(), "data": y.values.tolist()}
+            dataframe = pd.read_csv(y)
```

Comment:
Interesting that the type checker does not recognize that `y` is now a dataframe in this case.

Reply:
Yes these type checkers is not very good when the same variable is reassigned to a variable of a different type. Anyways, I think this change makes the code a little more readable so I just made the change